pkg/daemons/agent/agent.go

```diff
@@ -97,9 +99,13 @@ func startKubelet(cfg *config.Agent) error {
 	}
 	if cfg.RuntimeSocket != "" {
 		argsMap["container-runtime"] = "remote"
-		argsMap["container-runtime-endpoint"] = cfg.RuntimeSocket
 		argsMap["containerd"] = cfg.RuntimeSocket
 		argsMap["serialize-image-pulls"] = "false"
+		if strings.HasPrefix(argsMap["container-runtime-endpoint"], unixPrefix) {
+			argsMap["container-runtime-endpoint"] = cfg.RuntimeSocket
+		} else {
+			argsMap["container-runtime-endpoint"] = unixPrefix + cfg.RuntimeSocket
```

Comment:
shouldnt we just document that we that passing the endpoint should prefix with unix://

Reply:
no, because as noted here - only some things want it prefixed with unix://. cadvisor breaks if you prefix it.